src/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -180,14 +180,40 @@ private static SecurityStatusPal EncryptDecryptHelper(SafeDeleteContext security
 
         public static SecurityStatusPal ApplyAlertToken(ref SafeFreeCredentials credentialsHandle, SafeDeleteContext securityContext, TlsAlertType alertType, TlsAlertMessage alertMessage)
         {
-            // TODO (#12319): Not implemented.
+            // There doesn't seem to be an exposed API for writing an alert,
+            // the API seems to assume that all alerts are generated internally by
+            // SSLHandshake.
             return new SecurityStatusPal(SecurityStatusPalErrorCode.OK);
         }
 
         public static SecurityStatusPal ApplyShutdownToken(ref SafeFreeCredentials credentialsHandle, SafeDeleteContext securityContext)
         {
-            // TODO (#12319): Not implemented.
-            return new SecurityStatusPal(SecurityStatusPalErrorCode.OK);
+            SafeDeleteSslContext sslContext = ((SafeDeleteSslContext)securityContext);
+
+            // Unset the quiet shutdown option initially configured.
+            Interop.Ssl.SslSetQuietShutdown(sslContext.SslContext, 0);
+
+            IntPtr sslHandle = sslContext.SslContext.DangerousGetHandle();
```

Comment:
Why are we using DangerousGetHandle rather than using P/Invoke marshaling? If we stick with this, we either need to comment why it's safe or use DangerousAddRef/Release, too.